charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -73,34 +70,47 @@ spec:
-          - |
-            consul-api-gateway server \
-              -sds-server-host {{ template "consul.fullname" . }}-api-gateway-controller.{{ .Release.Namespace }}.svc \
-              -k8s-namespace {{ .Release.Namespace }} \
-              {{- if .Values.global.enableConsulNamespaces }}
-              {{- if .Values.apiGateway.consulNamespaces.consulDestinationNamespace }}
-              -consul-destination-namespace={{ .Values.apiGateway.consulNamespaces.consulDestinationNamespace }} \
-              {{- end }}
-              {{- if .Values.apiGateway.consulNamespaces.mirroringK8S }}
-              -mirroring-k8s=true \
-              {{- if .Values.apiGateway.consulNamespaces.mirroringK8SPrefix }}
-              -mirroring-k8s-prefix={{ .Values.apiGateway.consulNamespaces.mirroringK8SPrefix }} \
-              {{- end }}
-              {{- end }}
-              {{- end }}
-              -log-level {{ default .Values.global.logLevel .Values.apiGateway.logLevel }} \
+        - "/bin/sh"
+        - "-ec"
+        - |
+          consul-api-gateway server \
+            -sds-server-host {{ template "consul.fullname" . }}-api-gateway-controller.{{ .Release.Namespace }}.svc \
+            -k8s-namespace {{ .Release.Namespace }} \
+            {{- if .Values.global.enableConsulNamespaces }}
+            {{- if .Values.apiGateway.consulNamespaces.consulDestinationNamespace }}
+            -consul-destination-namespace={{ .Values.apiGateway.consulNamespaces.consulDestinationNamespace }} \
+            {{- end }}
+            {{- if .Values.apiGateway.consulNamespaces.mirroringK8S }}
+            -mirroring-k8s=true \
+            {{- if .Values.apiGateway.consulNamespaces.mirroringK8SPrefix }}
+            -mirroring-k8s-prefix={{ .Values.apiGateway.consulNamespaces.mirroringK8SPrefix }} \
+            {{- end }}
+            {{- end }}
+            {{- end }}
+            -log-level {{ default .Values.global.logLevel .Values.apiGateway.logLevel }} \
         volumeMounts:
-          {{- if .Values.global.tls.enabled }}
-          {{- if .Values.global.tls.enableAutoEncrypt }}
-          - name: consul-auto-encrypt-ca-cert
-          {{- else }}
-          - name: consul-ca-cert
-          {{- end }}
-            mountPath: /consul/tls/ca
-            readOnly: true
-          {{- end }}
+        {{- if .Values.global.tls.enabled }}
+        {{- if .Values.global.tls.enableAutoEncrypt }}
+        - name: consul-auto-encrypt-ca-cert
+        {{- else }}
+        - name: consul-ca-cert
+        {{- end }}
+          mountPath: /consul/tls/ca
+          readOnly: true
+        {{- end }}
+        - mountPath: /consul/login
+          name: consul-data
+          readOnly: true
+        {{- if .Values.global.acls.manageSystemACLs }}
+        lifecycle:
+          preStop:
+            exec:
+              command:
+              - "/bin/sh"
+              - "-ec"
+              - |
+                consul-k8s-control-plane consul-logout
```

Comment:
🚗 this binary is not available in this container. 🤔 I wonder how we should handle it here: maybe copy the consul binary?

Reply:
yeah, that makes sense

Reply:
I am using consul binary like [this](https://github.com/hashicorp/consul-k8s/pull/1070/files#diff-9b21e8a4c735f78620378655874a74263f5196a648e0e4c34e308150faf68f07R198).  Should I switch?

Reply:
I don't think so. In your case, the prestop hook will run in a container that's using the consul image so you're doing the right thing using the consul binary. 

For the api gateway, the image used is the gateway controller which doesn't have either consul-k8s-control-plane or the consul binaries, and so we need to copy either one.

Reply:
done! thanks for looking out!